examples/framesjs-starter/app/examples/base-mint-with-warps/pages/check.tsx

```diff
@@ -0,0 +1,128 @@
+import { kv } from "@vercel/kv";
+import { FrameButton, FrameContainer, FrameImage } from "frames.js/next/server";
+import { FrameState } from "frames.js/next/types";
+import { Card } from "../components/Card";
+import { ErrorFrame, MintFrame } from "../lib/responses";
+import signMintData from "../lib/signMint";
+import { FramePageProps, Session } from "../lib/types";
+import { isPrime } from "../lib/utils";
+import { SYNDICATE_API_KEY } from "../config";
+
+export async function CheckPage<T extends FrameState = FrameState>(
+  props: FramePageProps
+) {
+  const isFidPrime = isPrime(props.frameMessage.requesterFid);
+
+  if (isFidPrime) {
+    const fid = props.frameMessage.requesterFid;
+    let session = ((await kv.get(`session:${fid}`)) ?? {}) as Session;
+    const { address, transactionId, checks, retries } = session;
+    const totalChecks = checks ?? 0;
+    const totalRetries = retries ?? 0;
+
+    // If we've retried 3 times, give up
+    if (totalRetries > 2) {
+      return <ErrorFrame {...props}></ErrorFrame>;
+    }
+
+    // If we've checked 3 times, try to mint again
+    if (totalChecks > 2 && session.address) {
+      const { address } = session;
+      const sig = await signMintData({
+        to: address,
+        tokenId: 1,
+        fid,
+      });
+      const res = await fetch("https://frame.syndicate.io/api/mint", {
```

Comment:
this will take longer than 5s, no? whats the pattern for storing interrim values if this times out and never gets the kv set?

Reply:
The user will already be on the /check page, so if it times out they can just check again and it won't do both requests again